pkg/monitor/cluster/cache.go

```diff
@@ -59,3 +60,13 @@ func (mon *Monitor) listARODeployments(ctx context.Context) (*appsv1.DeploymentL
 	mon.cache.arodl, err = mon.cli.AppsV1().Deployments(pkgoperator.Namespace).List(ctx, metav1.ListOptions{})
 	return mon.cache.arodl, err
 }
+
+func (mon *Monitor) listAROClusters(ctx context.Context) (*arov1alpha1.ClusterList, error) {
```

Comment:
I am not sure this is a good idea. Do you see in this file how every list call has a comment saying it is an anti-pattern and should be avoided? Since we are not under strict business pressures to implement this I would ask we re-think this logic and do this correctly with pagination.

Reply:
Perhaps a limit parameter should be exposed in this function call here